File: Contains Duplicate/solution_sort.py
from typing import List


def contains_duplicate_sort(nums: List[int]) -> bool:
    """Return True if any duplicate exists using a sorting strategy."""

    sorted_nums = sorted(nums)

    for i in range(1, len(sorted_nums)):
        if sorted_nums[i] == sorted_nums[i - 1]:
            return True

    return False

File: Contains Duplicate/solution_sort.js
/**
 * Sorting-based approach to check for duplicates.
 * @param {number[]} nums
 * @returns {boolean}
 */
function containsDuplicateSort(nums) {
  const sorted = [...nums].sort((a, b) => a - b);

  for (let i = 1; i < sorted.length; i += 1) {
    if (sorted[i] === sorted[i - 1]) {
      return true;
    }
  }

  return false;
}

module.exports = { containsDuplicateSort };
